src/foremast/pipeline/construct_pipeline_block.py

```diff
@@ -70,13 +70,38 @@ def check_provider_healthcheck(settings, default_provider='Discovery'):
     return ProviderHealthCheck(providers=health_check_providers, has_healthcheck=has_healthcheck)
 
 
+def get_template_name(env, pipeline_type):
+    """Generates the correct template name based on pipeline type
+
+    Args:
+        env (str): environment to generate templates for
+        pipeline_type (str): Type of pipeline like ec2 or lambda
+    
+    Returns:
+        str: Name of template
+    """
+    if pipeline_type == 'ec2':
+        if env.startswith('prod'):
+            template_name = 'pipeline/pipeline_{}.json.j2'.format(env)
```

Comment:
It might be better to separate the string formats from the `.format()` method call. This would also require the use of keyword style format strings.
```python
if:
    template_name_format = '{env}'

template_name = template_name_format.format(env=env)
```
I think this looks a little cleaner and is slightly more extensible. It also might help us see how to collapse this `if` stack.